manta-trusted-setup/src/groth16/ppot/hashing.rs

```diff
@@ -0,0 +1,226 @@
+            .unwrap();
+        hash_to_group::<_, PpotDistribution, 64>(into_array_unchecked(hasher.0.finalize()))
+    }
+}
+
+/// A distribution to replicate random sampling as it was done
+/// during the Ppot ceremony, which used `rand v. 0.4`.
+#[derive(Copy, Clone, Default)]
+pub struct PpotDistribution;
+
+impl<P> Sample<PpotDistribution> for GroupAffine<P>
+where
+    P: SWModelParameters,
+    P::BaseField: Sample<PpotDistribution>,
+{
+    fn sample<R>(distribution: PpotDistribution, rng: &mut R) -> Self
+    where
+        R: RngCore + ?Sized,
+    {
+        loop {
+            let x = P::BaseField::sample(distribution, rng);
+            let greatest = bool::sample(distribution, rng);
+            if let Some(p) = Self::get_point_from_x(x, greatest) {
+                let p = p.scale_by_cofactor();
+                if !p.is_zero() {
+                    return p.into_affine();
+                }
+            }
+        }
+    }
+}
+
+impl Sample<PpotDistribution> for Fq {
+    fn sample<R>(distribution: PpotDistribution, rng: &mut R) -> Self
+    where
+        R: RngCore + ?Sized,
+    {
+        loop {
+            let mut tmp = BigInteger256::sample(distribution, rng);
+
+            // Mask away the unused bits at the beginning.
+            tmp.as_mut()[3] &= 0xffffffffffffffff >> ark_bn254::FqParameters::REPR_SHAVE_BITS;
+            if tmp < ark_bn254::FqParameters::MODULUS {
+                return Fp256::new(tmp);
+            }
+        }
+    }
+}
+impl Sample<PpotDistribution> for Fq2 {
+    fn sample<R>(distribution: PpotDistribution, rng: &mut R) -> Self
+    where
+        R: RngCore + ?Sized,
+    {
+        Self::new(Fq::sample(distribution, rng), Fq::sample(distribution, rng))
+    }
+}
+
+impl Sample<PpotDistribution> for BigInteger256 {
+    fn sample<R>(distribution: PpotDistribution, rng: &mut R) -> Self
+    where
+        R: RngCore + ?Sized,
+    {
+        BigInteger256([
+            u64::sample(distribution, rng),
+            u64::sample(distribution, rng),
+            u64::sample(distribution, rng),
+            u64::sample(distribution, rng),
+        ])
+    }
+}
+
+impl Sample<PpotDistribution> for u64 {
+    fn sample<R>(_: PpotDistribution, rng: &mut R) -> Self
+    where
+        R: RngCore + ?Sized,
+    {
+        ((rng.next_u32() as u64) << 32) | (rng.next_u32() as u64)
+    }
+}
+
+impl Sample<PpotDistribution> for bool {
+    fn sample<R>(_: PpotDistribution, rng: &mut R) -> Self
+    where
+        R: RngCore + ?Sized,
+    {
+        (rng.next_u32() as u8) & 1 == 1
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::groth16::{
+        kzg::{Accumulator, Proof},
+        ppot::{
+            kzg::PpotCeremony,
+            serialization::{read_kzg_proof, read_subaccumulator, Compressed, PpotSerializer},
+        },
+    };
+    use ark_std::println;
```

Comment:
Just use `println!`.

Reply:
The CI doesn't allow that, it says the macro doesn't exist when it's checking with no std.  Using `ark_std` used to work and now for some reason it doesn't, so I've instead flagged the tests with `#[cfg(feature = "std")]`. Is that a good solution?

Reply:
You should add `std` to the feature flags in the `dev-dependencies` for `manta-trusted-setup`.